src/Scene/Scene.js

```diff
@@ -138,18 +130,54 @@ Scene.prototype.scheduleUpdate = function scheduleUpdate(forceRedraw) {
 };
 
 Scene.prototype.update = function update() {
-    for (var l = 0; l < this.layers.length; l++) {
-        var layer = this.layers[l].node;
-
-        for (var sl = 0; sl < layer.children.length; sl++) {
-            var sLayer = layer.children[sl];
-
-            if (sLayer instanceof Quadtree) {
-                this.browserScene.updateQuadtree(this.layers[l], this.map.layersConfiguration, this.currentCamera());
-            } else if (sLayer instanceof MobileMappingLayer) {
-                this.browserScene.updateMobileMappingLayer(sLayer, this.currentCamera());
-            } else if (sLayer instanceof Layer) {
-                this.browserScene.updateLayer(sLayer, this.currentCamera());
+    this.camera.updateMatrixWorld();
+
+    // Browse Layer tree
+    const config = this.layersConfiguration;
+
+    // parent-free layers (hack for now)
+    const parentFreeLayers = config.getGeometryLayers().filter(l => !l.graftedOnId);
```

Comment:
I didn't want to use the parent/child semantics to make sure there's no confusion with the scenegraph parent/child relationship.
But I agree that this part must be simplified.